virtcontainers/device/manager/utils.go

```diff
@@ -35,3 +40,70 @@ func isVFIO(hostPath string) bool {
 func isBlock(devInfo config.DeviceInfo) bool {
 	return devInfo.DevType == "b"
 }
+
+func IsVFIOLargeBarSpaceDevice(hostPath string) (bool, error) {
+	if !isVFIO(hostPath) {
+		return false, nil
+	}
+	iommuDevicesPath := filepath.Join(config.SysIOMMUPath, filepath.Base(hostPath), "devices")
+	deviceFiles, err := ioutil.ReadDir(iommuDevicesPath)
+	if err != nil {
+		return false, err
+	}
+	// Pass all devices in iommu group
+	for _, deviceFile := range deviceFiles {
+		tokens := strings.Split(deviceFile.Name(), ":")
+		vfioDeviceType := config.VFIODeviceErrorType
+		if len(tokens) == 3 {
+			vfioDeviceType = config.VFIODeviceNormalType
+		} else {
+			tokens = strings.Split(deviceFile.Name(), "-")
+			if len(tokens) == 5 {
+				vfioDeviceType = config.VFIODeviceMediatedType
+			}
+		}
+		deviceFileName := filepath.Join(iommuDevicesPath, deviceFile.Name())
+		var isLarge bool
+		switch vfioDeviceType {
+		case config.VFIODeviceNormalType:
+			if isLarge, err = isLargeBarSpace(filepath.Join(deviceFileName, "resource")); err != nil {
+				return false, err
+			}
+			return isLarge, nil
+		case config.VFIODeviceMediatedType:
+			logrus.Warnf("Check sysfs resource for VFIODeviceMediatedType is not yet supported: %v", deviceFileName)
+		default:
+			logrus.Warnf("Incorrect tokens found while check sysfs resource: %s", deviceFileName)
+		}
+	}
+	return false, nil
+}
+
+func isLargeBarSpace(resourcePath string) (bool, error) {
+	buf, err := ioutil.ReadFile(resourcePath)
+	if err != nil {
+		return false, fmt.Errorf("failed to read sysfs resource: %v", err)
+	}
+	suffix := []string{"", "K", "M", "G", "T"}
+	for rIdx, line := range strings.Split(string(buf), "\n") {
```

Comment:
is possible to include an example (how the line looks like)

Reply:
Added.